lib/operations/create_indexes.js

```diff
@@ -2,36 +2,42 @@
 
 const Aspect = require('./operation').Aspect;
 const defineAspects = require('./operation').defineAspects;
-const OperationBase = require('./operation').OperationBase;
-const executeCommand = require('./db_ops').executeCommand;
+const CommandOperationV2 = require('./command_v2');
 const MongoError = require('../core').MongoError;
-const ReadPreference = require('../core').ReadPreference;
+const serverLacksFeature = require('../utils').serverLacksFeature;
 
-class CreateIndexesOperation extends OperationBase {
-  constructor(collection, indexSpecs, options) {
-    super(options);
-
-    this.collection = collection;
+class CreateIndexesOperation extends CommandOperationV2 {
+  /**
+   * @ignore
+   */
+  constructor(parent, name, indexSpecs, options, singular) {
+    super(parent, options);
+    this.name = name;
     this.indexSpecs = indexSpecs;
+    this.singular = singular;
   }
 
-  execute(callback) {
-    const coll = this.collection;
+  /**
+   * @ignore
+   */
+  execute(server, callback) {
+    const options = this.options;
     const indexSpecs = this.indexSpecs;
-    let options = this.options;
 
-    const capabilities = coll.s.topology.capabilities();
+    const collationNotSupported = serverLacksFeature(server, 'collation');
+    const commitQuorumNotSupported = serverLacksFeature(server, 'commitQuorum');
 
     // Ensure we generate the correct name if the parameter is not set
     for (let i = 0; i < indexSpecs.length; i++) {
+      // Did the user pass in a collation, check if our write server supports it
```

Comment:
I'm not sure why this was OK inside the `if (indexSpecs[i].name == null)` check below in the past, but I started getting errors and took a look, and it didn't make sense to me. Moving it outside, which I think makes more sense, also fixed the errors.